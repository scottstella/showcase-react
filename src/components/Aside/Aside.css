.aside {
  background-color: #c29591;
  width: 100%;
  padding: 30px;
}
