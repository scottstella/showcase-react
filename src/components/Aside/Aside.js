import React from "react";
import "./Aside.css";

export default function Aside() {
  return (
    <div className="aside">
      <ul>
        <li>
          <p>
            <strong>Developed:</strong>
          </p>
          <p>July 2022</p>
        </li>
        <li>
          <p>
            <strong>Showcasing</strong>
          </p>
          <p>React, Javascript, CSS</p>
        </li>
        <li>
          <p>
            <strong>Developed by:</strong>
          </p>
          <p>Scott Stella</p>
        </li>
      </ul>
    </div>
  );
}
