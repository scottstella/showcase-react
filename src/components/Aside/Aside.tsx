import React from "react";
import "./Aside.css";

export default function Aside() {
  return (
    <div className="aside">
      <ul>
        <li>
          <p>
            <strong>Developed:</strong>
          </p>
          <p>July 2022</p>
        </li>
        <li>
          <p>
            <strong>Showcasing</strong>
          </p>
          <p>
            React, Javascript, CSS, Supabase, Toastify, Font Awesome, Formik, Bruno
          </p>
        </li>
        <li>
          <p>
            <strong>Developed by:</strong>
          </p>
          <p>Scott Stella</p>
        </li>
      </ul>
    </div>
  );
}
