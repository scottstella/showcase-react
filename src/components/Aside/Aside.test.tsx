/// <reference types="vitest/globals" />
import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Aside from "./Aside";

describe("Aside", () => {
  it("matches snapshot", () => {
    const { container } = render(<Aside />);
    expect(container).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    render(<Aside />);
    expect(screen.getByRole("complementary")).toBeInTheDocument();
  });

  it("has the correct CSS class", () => {
    const { container } = render(<Aside />);
    const asideElement = container.querySelector("aside");
    expect(asideElement).toHaveClass("aside");
  });

  it("displays all expected content", () => {
    render(<Aside />);
    // Development date
    expect(screen.getByText("2022-2025")).toBeInTheDocument();

    // Technologies used
    expect(
      screen.getByText(/React, Javascript, CSS, Supabase, Toastify, Font Awesome, Formik, Bruno/i)
    ).toBeInTheDocument();

    // Developer name
    expect(screen.getByText("Scott Stella")).toBeInTheDocument();
  });

  it("displays all section headings", () => {
    render(<Aside />);
    expect(screen.getByText("Developed:")).toBeInTheDocument();
    expect(screen.getByText("Showcasing")).toBeInTheDocument();
    expect(screen.getByText("Developed by:")).toBeInTheDocument();
  });

  it("has the correct structure", () => {
    render(<Aside />);
    const listItems = screen.getAllByRole("listitem");
    expect(listItems).toHaveLength(3);
  });
});
