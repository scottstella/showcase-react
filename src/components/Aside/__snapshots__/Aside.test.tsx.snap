// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Aside > matches snapshot 1`] = `
<div>
  <aside
    class="aside"
    role="complementary"
  >
    <ul>
      <li>
        <p>
          <strong>
            Developed:
          </strong>
        </p>
        <p>
          2022-2025
        </p>
      </li>
      <li>
        <p>
          <strong>
            Showcasing
          </strong>
        </p>
        <p>
          React, Javascript, CSS, Supabase, Toastify, Font Awesome, Formik, Bruno
        </p>
      </li>
      <li>
        <p>
          <strong>
            Developed by:
          </strong>
        </p>
        <p>
          Scott Stella
        </p>
      </li>
    </ul>
  </aside>
</div>
`;
